tests/server/test_resourceservice.py

```diff
@@ -261,3 +256,77 @@ async def test_events_resource_without_dependencies(server, client, environment,
+    """
+    Test whether the `last_non_deploying_status` field is updated correctly when a deployment of a resource is done.
+    """
+    version = await clienthelper.get_version()
+    rvid_r1_v1 = ResourceVersionIdStr(f"std::File[agent1,path=/etc/file1],v={version}")
+    rvid_r2_v1 = ResourceVersionIdStr(f"std::File[agent1,path=/etc/file2],v={version}")
+    resources = [
+        {"path": "/etc/file1", "id": rvid_r1_v1, "requires": [], "purged": False, "send_event": False},
+        {"path": "/etc/file2", "id": rvid_r2_v1, "requires": [], "purged": False, "send_event": False},
+    ]
+    await clienthelper.put_version_simple(resources, version)
+
+    async def assert_status_fields(
+        r1_status: const.ResourceState,
+        r1_last_non_deploying_status: const.ResourceState,
+        r2_status: const.ResourceState,
+        r2_last_non_deploying_status: const.ResourceState,
+    ) -> None:
+        db_resources = await data.Resource.get_list(environment=environment)
+        rvid_to_resources = {res.resource_version_id: res for res in db_resources}
+        assert rvid_to_resources[rvid_r1_v1].status is r1_status
+        assert rvid_to_resources[rvid_r1_v1].last_non_deploying_status is r1_last_non_deploying_status
+        assert rvid_to_resources[rvid_r2_v1].status is r2_status
+        assert rvid_to_resources[rvid_r2_v1].last_non_deploying_status is r2_last_non_deploying_status
+
+    # All resources in available state
+    await assert_status_fields(
+        r1_status=const.ResourceState.available,
+        r1_last_non_deploying_status=const.ResourceState.available,
+        r2_status=const.ResourceState.available,
+        r2_last_non_deploying_status=const.ResourceState.available,
+    )
+
+    # Put R1 in deploying state
+    action_id_r1 = await resource_deployer.start_deployment(rvid=rvid_r1_v1)
+    await assert_status_fields(
+        r1_status=const.ResourceState.deploying,
+        r1_last_non_deploying_status=const.ResourceState.available,
+        r2_status=const.ResourceState.available,
+        r2_last_non_deploying_status=const.ResourceState.available,
+    )
+
+    # R1 finished deployment + R2 start deployment
+    await resource_deployer.deployment_finished(rvid=rvid_r1_v1, action_id=action_id_r1, status=const.ResourceState.deployed)
+    action_id_r2 = await resource_deployer.start_deployment(rvid=rvid_r2_v1)
+    await assert_status_fields(
+        r1_status=const.ResourceState.deployed,
+        r1_last_non_deploying_status=const.ResourceState.deployed,
+        r2_status=const.ResourceState.deploying,
+        r2_last_non_deploying_status=const.ResourceState.available,
+    )
+
+    # R1 start deployment + R2 skipped
+    action_id_r1 = await resource_deployer.start_deployment(rvid=rvid_r1_v1)
+    await resource_deployer.deployment_finished(rvid=rvid_r2_v1, action_id=action_id_r2, status=const.ResourceState.skipped)
```

Comment:
this test the new code path, but does this work on the old (resource_Action_update) code path as well?

Reply:
Good remark. I updated the test case to verify both code paths.